org-code-javabuilder/lib/src/main/java/org/code/javabuilder/UserProjectFileParser.java

```diff
@@ -21,20 +20,25 @@ public UserProjectFileParser() {
    * @throws UserFacingException
    * @throws UserInitiatedException
    */
-  public List<ProjectFile> parseFileJson(String json)
+  public void parseFileJson(
+      String json, List<JavaProjectFile> javaFileList, List<TextProjectFile> textFileList)
       throws UserFacingException, UserInitiatedException {
-    List<ProjectFile> fileList = new ArrayList<ProjectFile>();
     try {
       UserSourceData sourceData = this.objectMapper.readValue(json, UserSourceData.class);
       Map<String, UserFileData> sources = sourceData.getSource();
       for (String fileName : sources.keySet()) {
         UserFileData fileData = sources.get(fileName);
-        fileList.add(new ProjectFile(fileName, fileData.getText()));
+
+        if (fileName.indexOf(".java") > 0) {
```

Comment:
May want to update this to check if the file ends in java. i.e. I've seen people do myFile.java.txt before.